### PR TITLE
Override `createElement` to map svg elements to `react-native-svg` ones

### PR DESCRIPTION
**Pull request by Tug**
## Description
This PR reverts to using string versions of `svg` elements and introduces a custom `createElement` function for react-native which will be able to map those to native svg elements.
It will also be useful to have a custom `createElement` function in other case, such as having a failover for `<div>`, `<span>`... 

## How has this been tested?
To be tested along with https://github.com/wordpress-mobile/gutenberg-mobile/pull/184
WIP

## Types of changes
Build changes

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

File: packages/block-library/src/code/index.js [hunk 1]
```diff
@@ -20,7 +16,7 @@ export const settings = {
 
 	description: __( 'Add text that respects your spacing and tabs -- perfect for displaying code.' ),
 
-	icon: <SVG viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><Path d="M0,0h24v24H0V0z" fill="none" /><Path d="M9.4,16.6L4.8,12l4.6-4.6L8,6l-6,6l6,6L9.4,16.6z M14.6,16.6l4.6-4.6l-4.6-4.6L16,6l6,6l-6,6L14.6,16.6z" /></SVG>,
+	icon: <svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M0,0h24v24H0V0z" fill="none" /><path d="M9.4,16.6L4.8,12l4.6-4.6L8,6l-6,6l6,6L9.4,16.6z M14.6,16.6l4.6-4.6l-4.6-4.6L16,6l6,6l-6,6L14.6,16.6z" /></svg>,
```
Comment (gziolo) [inline packages/block-library/src/code/index.js]: Not sure about `path`, `g` and rest, but at least `SVG` needs to stay as it adds some accessibility features behind the scenes. It's expected to have it everywhere.

**Comment by gziolo** [comment]
Did you make it work? Changes in both PRs look reasonable. I will try to test it on Friday.

**Comment by Tug** [comment]
It works, but still needs some cleanup on the mobile PR (around how we import/build modules).
Also tests are failing + some conflicts already :)

File: packages/block-library/src/code/index.js (hunk 1 above)
Comment (Tug) [reply packages/block-library/src/code/index.js]: Couldn't we just map `svg` to our `SVG` then (instead of the one from react-native-svg`)?

**Comment by gziolo** [comment]
Let me know when it is fully working, I will ping a few more developers to look closer into it.

The solution we have at the moment helped us to move forward, but I'm still not 100% confident it's bulletproof in the long run. Having this Eslint rule:
https://github.com/WordPress/gutenberg/blob/master/.eslintrc.js#L169-L183
makes it a bit problematic to use `SVG` and family on mobile inside `save` method. We don't do it at the moment so it's non issue, but I'm worried that we can't use any RN component inside save as they won't serialize to HTML.

**Comment by gziolo** [comment]
@Tug do you plan to continue working on this PR? It got very out of date :(

**Comment by Tug** [comment]
It's not a priority at the moment no 👍
Will reopen if we want to come back to this again.

**Pull request closed without merging**